source/org/zfin/uniprot/NcbiMatchThroughEnsemblTask.java

```diff
@@ -0,0 +1,363 @@
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.sql.SQLException;
+
+import java.net.URL;
+import java.util.List;
+import java.util.zip.GZIPInputStream;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+
+
+/**
+ *
+ * If ncbi has a link to us, but we don't reciprocate, we check for shared ensembl IDs.
+ * This is a supplementary check to the NCBI Load (see ZFIN-8517)
+ *
+ * Description from ticket:
+ * Our current NCBI Load is missing some potential matches between ZFIN and NCBI Gene records.  Our NCBI Load script matches on RNA sequence IDs and, in cases where there is no RNA sequence ID, there is no way to match records.
+ * NCBI does annotate links to ZFIN genes and I’ve found that the associations they create are quite good.  So NCBI may have a correct match to a ZFIN gene, but we don’t have a reciprocal link.
+ * My goal with this report is to identify potential ZFIN/NCBI Gene matches that are supported by additional data, make those associations and pull data from NCBI Gene (including RefSeq IDs) onto ZFIN gene pages as is done with the current NCBI load.  The addition of this data adds to the robustness of ZFIN gene records and adding additional RefSeqs will allow us to associate more UniProt IDs and the resulting UniProt data.
+ * A reasonable method to do this would be to find all NCBI Gene records that have a link to ZFIN. Ignore the NCBI Gene records to which we already have links.  For those NCBI Gene records with links to ZFIN that don’t have reciprocal links from ZFIN to NCBI, obtain the Ensembl gene ID in the NCBI Gene records and match those against Ensembl gene IDs on ZFIN gene records.  This list would contain links that NCBI has made to ZFIN and are supported by a shared Ensembl ID.
+ * Finally, identify any instances where the Ensembl ID from NCBI hits a ZFIN gene that already contains a link to an NCBI Gene record (there is some sort of conflict that needs to be resolved).
+ *
+ * Call with first argument being the input file (https://ftp.ncbi.nlm.nih.gov/gene/DATA/GENE_INFO/Non-mammalian_vertebrates/Danio_rerio.gene_info.gz)
+ * Alternatively, set the environment variable NCBI_FILE_URL to the input file.
+ *
+ */
+public class NcbiMatchThroughEnsemblTask extends AbstractScriptWrapper {
+    private static final String CSV_FILE = "ncbi_matches_through_ensembl.csv";
+    private static final String DEFAULT_INPUT_FILE_URL = "https://ftp.ncbi.nlm.nih.gov/gene/DATA/GENE_INFO/Non-mammalian_vertebrates/Danio_rerio.gene_info.gz";
+    private String inputFileUrl;
+
+    public static void main(String[] args) {
+        NcbiMatchThroughEnsemblTask task = new NcbiMatchThroughEnsemblTask();
+
+        try {
+            task.runTask(args);
+        } catch (IOException e) {
+            System.err.println("IOException Error while running task: " + e.getMessage());
+            e.printStackTrace();
+            System.exit(1);
+        } catch (SQLException e) {
```

Comment:
You can collapse the two exceptions into one as in both cases the system exits the program.